solidity/contracts/hooks/ArbitrumMessageHook.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+import {IArbitrumMessageHook} from "../interfaces/hooks/IArbitrumMessageHook.sol";
+import {ArbitrumISM} from "../isms/native/ArbitrumISM.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+
+// ============ External Imports ============
+import {IInbox} from "@arbitrum/nitro-contracts/src/bridge/IInbox.sol";
+
+/**
+ * @title ArbitrumMessageHook
+ * @notice Message hook to inform the Arbitrum ISM of messages published through
+ * the native Arbitrum bridge.
+ */
+contract ArbitrumMessageHook is IArbitrumMessageHook {
+    // ============ Constants ============
+
+    // Domain of chain on which the arbitrum ISM is deployed
+    uint32 public immutable destinationDomain;
+    // Arbitrum ISM to verify messages
+    ArbitrumISM public immutable ism;
+    // Arbitrum's inbox used to send messages from L1 -> L2
+    IInbox public immutable inbox;
+
+    uint128 internal constant MAX_GAS_LIMIT = 26_000;
+    uint128 internal constant MAX_GAS_PRICE = 1e9;
+
+    // ============ Constructor ============
+
+    constructor(
+        uint32 _destinationDomain,
+        address _inbox,
+        address _ism
+    ) {
+        require(_inbox != address(0), "ArbitrumHook: invalid inbox address");
+        require(_ism != address(0), "ArbitrumHook: invalid ISM address");
```

Comment:
I think we use `Address.isContract` elsewhere

Reply:
fixed with _onlyContract()